components/user/EventRow.tsx

```diff
@@ -0,0 +1,296 @@
+import { Fragment, ReactElement, useState } from 'react'
+import clsx from 'clsx'
+
+import useClipboard from 'react-use-clipboard'
+import {
+  format,
+  isBefore,
+  isAfter,
+  parseISO,
+  eachWeekOfInterval,
+} from 'date-fns'
+import enUS from 'date-fns/locale/en-US'
+
+import ActivityBlockMined from 'components/icons/ActivityBlockMined'
+import ActivityBugReported from 'components/icons/ActivityBugReport'
+import ActivityPullRequest from 'components/icons/ActivityPullRequest'
+import ActivityCopy from 'components/icons/ActivityCopy'
+import ActivityCommunityContribution from 'components/icons/ActivityCommunityContribution'
+import ActivitySocial from 'components/icons/ActivitySocial'
+import ChevronRight from 'components/icons/ChevronRight'
+
+import {
+  EventType,
+  ApiEvent,
+  ApiEventMetadata,
+  ApiEventMetadataBlockMined,
+  ApiEventMetadataWithLink,
+} from 'apiClient/index'
+
+import styles from './EventRow.module.css'
+
+interface IconText {
+  icon: ReactElement
+  text: string
+}
+export function displayEventType(type: EventType): IconText {
+  const text =
+    type === 'BLOCK_MINED'
+      ? 'Mined a block'
+      : type === 'BUG_CAUGHT'
+      ? 'Reported a bug'
+      : type === 'COMMUNITY_CONTRIBUTION'
+      ? 'Contributed to the community'
+      : type === 'PULL_REQUEST_MERGED'
+      ? 'Submitted a Pull Request'
+      : type === 'SOCIAL_MEDIA_PROMOTION'
+      ? 'Promoted testnet'
+      : type
+  const icon =
+    type === 'BLOCK_MINED' ? (
+      <ActivityBlockMined />
+    ) : type === 'BUG_CAUGHT' ? (
+      <ActivityBugReported />
+    ) : type === 'COMMUNITY_CONTRIBUTION' ? (
+      <ActivityCommunityContribution />
+    ) : type === 'PULL_REQUEST_MERGED' ? (
+      <ActivityPullRequest />
+    ) : type === 'SOCIAL_MEDIA_PROMOTION' ? (
+      <ActivitySocial />
+    ) : (
+      type
+    )
+  return { icon, text }
+}
+export type EventRowProps = {
+  id: number
+  type: EventType
+  occurred_at: string
+  points: number
+  metadata?: ApiEventMetadata
+}
+
+const makeLinkForEvent = (type: EventType, metadata?: ApiEventMetadata) => {
+  if (!metadata) return ''
+  if ('hash' in metadata && type === EventType.BLOCK_MINED) {
+    return `https://explorer.ironfish.network/blocks/${metadata.hash}`
+  }
+  return (metadata as ApiEventMetadataWithLink).url
+}
+
+type CopyableHashProps = {
+  hash: string
+}
+
+enum CopyState {
+  IDLE = 'copy_idle',
+  OVER = 'copy_over',
+  COPIED = 'copy_copied',
+  RESETTING = 'copy_resetting',
+}
+
+const CopyableHash = ({ hash }: CopyableHashProps) => {
+  const [, $setCopied] = useClipboard(hash, {
+    successDuration: 1000,
+  })
+  const [$hover, $setHover] = useState<CopyState>(CopyState.IDLE)
+  const abbrevHash = hash.slice(45, Infinity)
```

Comment:
I don't think the `Infinity` is needed here, the second parameter in `slice` is optional

Reply:
This may have been resolved by mistake, looks like it's still here?

Reply:
Oh good catch, my apologies.